openstack_hypervisor/hooks.py

```diff
@@ -29,9 +29,11 @@
 from jinja2 import Environment, FileSystemLoader, Template
 from snaphelpers import Snap
 
+from openstack_hypervisor import utils
 from openstack_hypervisor.log import setup_logging
 
 UNSET = ""
+LOCAL_IP_BY_DEFAULT_ROUTE = utils.get_local_ip_by_default_route()
```

Comment:
I'm not a huge fan of global execution in this way - see below

Reply:
Actually see here - if we just pass the function as the value in the dict then in _update_default_config we can use something like

```
if callable(default):
  default = default()
```

this can be applied to socket.fqdn as well.

Reply:
this avoids execution of functions during the construction of the dict as the module is imported